jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/Server.java

```diff
@@ -539,12 +539,31 @@ protected void doStart() throws Exception
             if (getStopAtShutdown())
                 ShutdownThread.register(this);
 
-            //Register the Server with the handler thread for receiving
-            //remote stop commands
-            ShutdownMonitor.register(this);
+            // Look for the old-school (now deprecated) ShutdownMonitor
+            // First we look to see if anyone has called ShutdownMonitor.getInstance() yet.
+            //noinspection removal
+            ShutdownMonitor shutdownMonitor = ShutdownMonitor.INSTANCE.get();
+            //noinspection removal
+            if (shutdownMonitor != null && shutdownMonitor.getPort() != (-1))
+            {
+                addBean(shutdownMonitor);
+            }
 
-            //Start a thread waiting to receive "stop" commands.
-            ShutdownMonitor.getInstance().start(); // initialize
+            if (getBean(ShutdownService.class) == null)
+            {
+                // Attempt to create the default ShutdownService based on historical System Properties.
+                ShutdownService shutdownService = ShutdownService.createFromHistoricalSystemProperties();
+                if (shutdownService != null)
+                {
+                    addBean(shutdownService);
+                }
```

Comment:
Rather than doing this here, it should be in a method `ShutdownMonitor.getInstanceIfConfigured()` that you can use above.   I.e. if there a system properties, then one static instance is created, rather than a new one here.

Reply:
@joakime The problem with the current approach is that if the system properties are set and no ShutdownService is set, then creating multiple Server instances will result in an exception as they will each try to create a ShutdownService from the system properties and they will clash on the port.

If the system properties are set, there can only be a single instance in the JVM.   The ShutdownMonitor is that instance.  It is just a ShutdownService that is configured from the system properties, only if they are set and a method like my suggested `ShutdownMonitor.getInstanceIfConfigured()`.

While we support the system properties, we can never remove the singleton, as they both are one per JVM.  So they can both be deprecated and both will be removed at the same time.